pkg/resources/statefulsets/prometheus.go

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2020, MinIO, Inc.
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+package statefulsets
+
+import (
+	miniov1 "github.com/minio/operator/pkg/apis/minio.min.io/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// prometheusMetadata returns the object metadata for Prometheus pods. User
+// specified metadata in the spec is also included here.
+func prometheusMetadata(t *miniov1.Tenant) metav1.ObjectMeta {
+	meta := metav1.ObjectMeta{
+		Labels:      t.Spec.Prometheus.Labels,
+		Annotations: t.Spec.Prometheus.Annotations,
+	}
+	if meta.Labels == nil {
+		meta.Labels = make(map[string]string)
+	}
+	for k, v := range t.PrometheusPodLabels() {
+		meta.Labels[k] = v
+	}
+	return meta
+}
+
+// prometheusSelector returns the prometheus pods selector
+func prometheusSelector(t *miniov1.Tenant) *metav1.LabelSelector {
+	return &metav1.LabelSelector{
```

Comment:
shouldn't this include user supplied pod labels from `t.Spec.Prometheus.Labels`?

Reply:
Added now.